### DISCUSSION TITLE
Data Dictionary

### DISCUSSION THREAD

**Discussion by clhambone** [trimble-oss/dba-dash#752] [General]
I know this is a Free solution and I cannot tell you how much I appreciate this tool.  It is tremendous for open source.  I am trying to add in some custom tables to see if they will stay in place after using the upgrade functionality in the config tool.  That being said, I was curious if you guys have considered releasing any sort of data dictionary for the application.  Thanks

**Comment by DavidWiseman** [comment]
Maybe - I'd like to improve the documentation anyway.  If anyone is interested in contributing, the dbadash.com website is [also open source](https://github.com/trimble-oss/dba-dash-docs).

User customizations should no longer be dropped when upgrading unless there is a naming conflict. The next version will introduce a UserData schema where I will encourage users to put their own data to avoid potential conflicts.  Or creating your own unique schema should be OK.  It's always worth having backups of the repository DB in place should you need to roll back for some reason.

The last release added custom reports.  These will get better in the next version with drill-downs and other types of links.  Also, custom collections will be introduced where you can get the DBA Dash service to run a stored procedure on a schedule and import it into the repository DB.  DBA Dash will generate a lot of the required T-SQL code for you to review and tweak if required.  It's going to be easy to get the tool to run something like sp_BlitzIndex on a schedule and have some reporting on it - or whatever community or homegrown script.

**Comment by clhambone** [comment]
I am not much of a programmer but I was curious if I found ways to do a
query. For example, looking for jobs that run longer than averages. How
would I submit that back to you guys? And then maybe y'all could
incorporate something like that into the Application?

**Comment by DavidWiseman** [comment]
The next version will include a "Script Report" feature to make it easy for people to share their custom report creations with the community.  

![image](https://github.com/trimble-oss/dba-dash/assets/33157668/8f99155f-c363-407d-919e-b26e8c0b7a9f)

This script is your stored procedure plus any customization/formatting done to the report in the app.  

Initially, I thought these discussion pages might be a good place for people to share their report creations.  I added a tag "Custom Report" that can be used to make it easier for people to find reports.  It's great that you are interested in sharing your report - I'm hoping other users will also be interested in doing this.  I want to make it easy for people to do custom data collections and custom reports and to share with the community - the next version will make that better and easier.

Adding as a standard part of the app is a bit more work at the moment and requires some C# skills.  I'll probably use the custom collection and custom reporting myself to prototype new features.  I'm also considering introducing system reports at some point that will just re-use the work I've done on user reports.  At that point converting a user report to a system report should be low effort.

Thanks
